sbom-utility-scripts/scripts/base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -92,6 +94,13 @@ def parse_args():
         description="Updates the sbom file with base images data based on the provided files"
     )
     parser.add_argument("--sbom", type=pathlib.Path, help="Path to the sbom file", required=True)
+    parser.add_argument(
+        "--sbom-type",
+        choices=["spdx", "cyclonedx"],
+        default="cyclonedx",
+        help="Type of the sbom file",
+        required=True,
+    )
```

Comment:
Can we auto-detect this? Requiring the type to be explicitly specified will cause problems

Reply:
I guess we can, but what if you want to for example work with cyclonedx but you receive spdx. Idea is that we switch whole pipeline to spdx with single parameter, so unexpected format should produce a failure

Reply:
> Idea is that we switch whole pipeline to spdx with single parameter

That's not really possible. Konflux proposes the initial pipeline to the user's repo in a PR, but after that, the pipeline is fully in the user's control. "Switch to spdx with a single parameter" would be more like "every Konflux user needs to set this parameter for all the tasks in all their pipelines."

Think of it this way - the main unit that Konflux produces is the Task, not the Pipeline. Interoperability and backwards compatibility needs to be handled on the Task level

Reply:
I was told users basically creates custom "PipelineRuns" but the pipelines and task definitions are take from the main repo. So idea is to have default set to cyclonedx and after we're ready to switch to spdx we can change the default to spdx. Users don't even needs to know there's some parametr which controls sbom format

Reply:
Yes, when a user onboards to Konflux, their pipeline is generated based on the state of build-definition at the time. But when the pipeline definition in build-definition changes, the pipeline in users' repos does not.

So the switch to SPDX will require a mass migration

Reply:
But since we'll likely need to do that mass migration no matter what, maybe requiring a param is not so bad. It is still worse than making the buildah task work regardless of input SBOM format, but if it would add that much extra complexity, we can rely on a param

Reply:
auto-detection would be great, but like @chmeliik  is saying maybe having the parameter would be sufficient solution here. Should we discuss this in the build team @chmeliik so we can resolve this?

Reply:
I'd like input from @midnightercz on the complexity of making it smarter. It feels trivial.